pkg/cmd/boot/boot_upgrade.go

```diff
@@ -0,0 +1,268 @@
+package boot
+
+import (
+	"fmt"
+	"github.com/jenkins-x/jx/pkg/cmd/helper"
+	"github.com/jenkins-x/jx/pkg/cmd/opts"
+	"github.com/jenkins-x/jx/pkg/cmd/templates"
+	"github.com/jenkins-x/jx/pkg/config"
+	"github.com/jenkins-x/jx/pkg/gits"
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/jenkins-x/jx/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// BootUpgradeOptions options for the command
+type BootUpgradeOptions struct {
+	*opts.CommonOptions
+	Dir string
+}
+
+var (
+	bootUpgradeLong = templates.LongDesc(`
+		This command creates a pr for upgrading a jx boot gitOps cluster, incorporating changes to boot
+        config and version stream ref
+`)
+
+	bootUpgradeExample = templates.Examples(`
+		# create pr for upgrading a jx boot gitOps cluster
+		jx boot upgrade
+`)
+)
+
+// NewCmdBootUpgrade creates the command
+func NewCmdBootUpgrade(commonOpts *opts.CommonOptions) *cobra.Command {
+	options := &BootUpgradeOptions{
+		CommonOptions: commonOpts,
+	}
+	cmd := &cobra.Command{
+		Use:     "boot upgrade",
+		Aliases: []string{"bu"},
+		Short:   "Upgrades jx boot config",
+		Long:    bootUpgradeLong,
+		Example: bootUpgradeExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			options.Cmd = cmd
+			options.Args = args
+			err := options.Run()
+			helper.CheckErr(err)
+		},
+	}
+	cmd.Flags().StringVarP(&options.Dir, "dir", "d", ".", "the directory to look for the Jenkins X Pipeline and requirements")
+	return cmd
+}
+
+// Run runs this command
+func (o *BootUpgradeOptions) Run() error {
+	reqsVersionStream, err := o.requirementsVersionStream(o.Dir)
+	if err != nil {
+		return err
+	}
+
+	upgradeVersionSha, err := o.upgradeAvailable(reqsVersionStream.URL, reqsVersionStream.Ref, "master")
+	if err != nil {
+		return err
+	}
+	if upgradeVersionSha == "" {
+		return nil
+	}
+
+	err = o.checkoutNewBranch(o.Dir, "upgrade_branch")
```

Comment:
Do we delete this branch after the update is applied? I think it will probably fail second time when an update is created if the branch already exists.